### DISCUSSION TITLE
What's the exact purpose of this generated method?

### DISCUSSION THREAD

**Discussion by vsajip** [congo-cc/congo-parser-generator#164] [Q&A] [answered]
https://github.com/congo-cc/congo-parser-generator/blob/7a30fcf1d0bbc78906da911690c877ecfd2da0f6/src/templates/java/Parser.java.ftl#L281-L287

Sometimes, it's the only thing referencing `hitFailure` (for example, in the C# `PPDirectiveLine` parser, but could also be true for other simple languages).

**Comment by revusky** [comment, marked as answer]
Well, I guess you could say it exists to generalize the failure of an assertion, i.e. the routine works both when in lookahead and in parsing. I mean, if `currentLookaheadToken == null` then we are *not* in a lookahead, i.e. we are parsing, so it just throws the exception. If we are in a lookahead routine, we set the `hitFilaure` flag to true, so that the lookahead routine we're in will fail at the first opportunity.

**Reply by revusky** [reply]
I hope what I just wrote is clear. In any case, do feel free to add comments to the code.

**Reply by vsajip** [reply]
Yes, that makes sense - thanks.
